semgrep-core/src/parsing/tree_sitter/Parse_hcl_tree_sitter.ml

```diff
@@ -85,16 +85,38 @@ let map_numeric_lit (env : env) (x : CST.numeric_lit) : literal =
 
 let map_template_literal (env : env) (xs : CST.template_literal) :
     string wrap option =
-  (* each character is a separate template_literal_chunk, so we need
-   * to merge them in a single string.
-   *)
-  let xs = List.map (str env (* template_literal_chunk *)) xs in
   match xs with
   | [] -> None
-  | (s1, t1) :: ys ->
-      let str = List.map fst ys |> String.concat "" in
-      let toks = List.map snd ys in
-      Some (s1 ^ str, PI.combine_infos t1 toks)
+  | ((x_pos, x_str) as x) :: xs ->
+      (* each character is a separate template_literal_chunk, so we need
```

Comment:
This is an old comment, but just noting that I'm confused by what "each character is a separate template_literal_chunk" means